### PR TITLE
feat: S3 file picker as a drawer

### PR DESCRIPTION
**Pull request by gbouv**
https://github.com/windmill-labs/windmill/assets/18011812/70e4aa75-5ae8-49da-b12a-d7a33505e33f

### CODE REVIEW

**Comment by gbouv** [comment]
I don't quite get why I had to change NodeJS.Timer to NodeJS.Timeout (which is one of its implementation). Nothing in my PR bumps a dependency or anything that can change a dependendant library. For reference, here is the failing build: https://github.com/windmill-labs/windmill/actions/runs/6914286590/job/18811978315?pr=2640.
Happy to revert that commit if that's problematic

File: frontend/src/routes/(root)/(logged)/runs/[...path]/+page.svelte [hunk 1]
```diff
@@ -33,7 +33,7 @@
 	import RunsQueue from '$lib/components/runs/RunsQueue.svelte'
 
 	let jobs: Job[] | undefined
-	let intervalId: NodeJS.Timer | undefined
+	let intervalId: NodeJS.Timeout | undefined
```
Comment (gbouv) [inline frontend/src/routes/(root)/(logged)/runs/[...path]/+page.svelte:36]: See https://github.com/windmill-labs/windmill/pull/2640#issuecomment-1817517261

File: frontend/src/lib/components/apps/editor/RecomputeAllComponents.svelte [hunk 2]
```diff
@@ -12,7 +12,7 @@
 	const appEditorContext = getContext<AppEditorContext>('AppEditorContext')
 
 	let loading: boolean = false
-	let timeout: NodeJS.Timer | undefined = undefined
+	let timeout: NodeJS.Timeout | undefined = undefined
```
Comment (gbouv) [inline frontend/src/lib/components/apps/editor/RecomputeAllComponents.svelte]: See https://github.com/windmill-labs/windmill/pull/2640#issuecomment-1817517261

File: backend/windmill-api/src/job_helpers.rs [hunk 3]
```diff
@@ -219,11 +238,157 @@ async fn list_stored_datasets(
         }
     }
 
+    let file_count = stored_datasets.len();
+    #[cfg(not(feature = "enterprise"))]
+    if stored_datasets.len() > 10 {
+        stored_datasets = vec![];
```
Comment (rubenfiszel) [inline backend/windmill-api/src/job_helpers.rs]: should return an error there

File: frontend/src/lib/components/ArgInput.svelte [hunk 4]
```diff
@@ -385,6 +398,30 @@
 				<span class="text-2xs text-tertiary">Loading resource types...</span>
 			{:else if inputCat == 'resource-object' && (resourceTypes == undefined || (format.split('-').length > 1 && resourceTypes.includes(format.substring('resource-'.length))))}
 				<ObjectResourceInput {disablePortal} {format} bind:value {showSchemaExplorer} />
+			{:else if inputCat == 'resource-object' && format.split('-').length > 1 && format.replace('resource-', '') == 'dataset'}
```
Comment (rubenfiszel) [inline frontend/src/lib/components/ArgInput.svelte]: what about we replace dataset with s3object ?

File: frontend/src/lib/components/ArgInput.svelte [hunk 5]
```diff
@@ -385,6 +398,30 @@
...
+						color="light"
+						size="xs"
+						btnClasses="mt-1"
+						on:click={() => {
+							s3FilePicker?.open?.()
+						}}
+						id="arg-input-file-picker"
+						startIcon={{ icon: Pipette }}
+					>
+						Choose a dataset from the catalog
```
Comment (rubenfiszel) [inline frontend/src/lib/components/ArgInput.svelte]: an object

**Pull request merged**